### PR TITLE
Reality checks Rivtech guns and 8x40mm.

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
Balance "revamps the 8x40mm caseless round into an above-average combat cartridge and a quality recreational civilian round, Formulating new lore to Illustrate the role Rivtech had to play in the pre-cataclysm, and reworking their firearms."
#### Purpose of change
Rivtech's 8x40mm caseless munitions and associated weapon systems have been an integral part of Cataclysm for several years, representing some of the game's highest-quality firepower, excluding explosives. Given the potency of these cartridges, it's only logical for them to be deeply embedded in the game's lore and world-building. However, Rivtech equipment currently occupies an undefined gray area where it is effective as end-game gear but lacks justification for its potency, in addition, in my view, to lacking tangibly distinct characteristics that capitalize on their caseless nature.
The primary issue with these guns is not their power but rather the fact that they function as exceptional wonder weapons without any established lore to justify why their rounds are significantly superior to other rifle cartridges. At present, 8x40mm can be classified as a full-power rifle round that can be effectively chambered in anything from semi-automatic handguns to heavy machineguns and be effective at any task it’s put to, whether as a sniper’s cartridge or for CQC room clearing. The cartridge has astounding armor penetration numbers by any metric, with regular 8x40mm caseless possessing 18 armor penetration, exceeding most sniper rifle rounds and standing on par with 5.7x28mm, a PDW cartridge expressly designed for defeating ballistic armor. 8x40mm JHP is comparable to.300 Winchester Magnum, while FMJ rounds outperform.50 BMG ball ammo.
Furthermore, the guns themselves boast statistics that surpass most other weapons, seemingly without any justifiable reason. For instance, the RM11B scout rifle exhibits nine times the accuracy of the M2010 sniper rifle, even without considering the dispersion reductions from the weapon's scope. Similarly, the RM88 battle rifle is triple as accurate as the M2010, the RM614 LMG is nearly 1/3 as accurate, the RM298 HMG is equally accurate, and the RM2000 submachine gun is 1/3 less accurate. Weights and volumes also vary greatly, with the Rivtech RM 614, for instance, being only 1/3 the size of the M249.
In summary, the current Rivtech guns suffer from being powerful all-in-one weapons without a compelling reason for their exceptional performance across the board. They lack defining traits beyond their name, descriptions, and rare ammunition, behaving similarly to an M4 but with inflated numbers. There's no lore to justify why these caseless arms succeeded where other similar weapons like the H&K G11, VEC-91, and VAG-73 failed due to ammunition cook-off issues, impractical complexity, or general expensiveness. Furthermore, the design choices, which I will outline later, raise questions from both a practical and a development perspective.
The solution lies not in disregarding these incongruities and leaving the guns as they are, nor in unreasonably diminishing their performance to adhere strictly to realism. The latter approach would strip the game of compelling end-game firearms and overlook the world-building potential that these weapons offer. Conversely, ignoring the blatant lack of realism in a game that aims for authenticity in all aspects would be, in my opinion, a mistake. A balanced approach is required, blending realism and game convenience while deepening the integration of the guns, Rivtech, and caseless technology into player-facing lore to make the weapons interesting, effective, and moderately plausible.
#### New lore andwWorldbuilding
The cornerstone of this PR is working towards introducing lore and related world-building elements to flush out Rivtech as a company, their products’ development, and the relations they had with the world in the pre-Cataclysm. This includes vaguely pinning down the origins of interest in caseless rifle technology, a developmental concept that has been resting dead in the water for several decades by this point, with the LSAT program being the only serious investigation into caseless and telescoped rounds conducted in the west, barring failed commercial entries like the VEC-91. Given that the LSAT program eventually culminated in Textron Systems telescoped entry for the NGSW trials and no other serious delve into caseless weapons has been made since then, it is doubtful to assume that the US would have had an interest in producing an additional caseless rifle concept atop of the aforementioned program. As such, the working lore maintains that the latest, functional probing of such technology hails from pre-2018 Chinese experiments.
The lore leaves it deliberately vague as to why China gained an interest in developing a domestically produced caseless rifle. It could be anything from them experimenting with their own portal explorations and retrieving the guns from a parallel dimension, encountering a team sent from a parallel Earth on their own portal exploration armed with the weapons, or simply wishing to work on developing a better combat rifle in step with the West. The latter is not too unlikely considering the cold war present during CDDA’s timeline and is more than likely the most plausible, lore-friendly scenario. Regardless of the how and why, this allows us to incorporate practical, modern caseless weapons into the game without having to fluff up justification on why anyone in the US wished to explore this avenue and what they did to make caseless ammo practical: removing overheating issues and cookoff issues. In addition, this further expands upon viewable Chinese-American relations, adding some more information to happenings in the pre-Cataclysm beyond America’s portal shenanigans.
Rivtech itself enters the picture during the opening days of the US Army’s NGSW program. It was a very lucrative contract, and it is reasonable to assume that Rivtech, already likely an advanced defense contractor possibly working in part to develop components for experimental combat exoskeletons, would wish to have its own, increased slice of the American military pie. Through one way or another, the exact details of which are deliberately vague, Rivtech acquired the information that China had developed effective caseless rifle prototypes and, preferring to acquire an advanced weapon rather than diverting large resources from their current projects and spending man hours inventing a rifle from scratch, sought to acquire these prototypes. The exact details of their acquisition are left rather vague, with some in-game sources claiming that they simply clandestinely smuggled examples of these weapons from Chinese territories or even leaked sensitive information regarding the mechanics of powered combat exoskeletons to the Chinese in exchange for examples of the guns. Regardless of the details, Rivtech ends up with caseless prototypes in time to reverse engineer them and submit their version of the rifle and adapted machinegun for the NGSW program.
Ultimately, the caseless guns get rejected due to their complexity and high production cost for both weapons and munitions, and, as in the real world, the XM7 and XM250 get adopted instead. Like other competitors in the program whose designs failed to pass, Rivtech seeks to commercialize the project, offering its combat rifle and machinegun for paramilitary and security contractor use and selling semi-automatic versions of the guns as expensive civilian weapons. It is in this latter vein that they also developed the 20x66 caseless shotgun cartridge.
At some point shortly before the Cataclysm, Xedra makes note of how advanced the caseless guns seem to be and, growing curious, engages in a game of interdepartmental telephone. This brings to the fore the realization that nobody has a clue how and where Rivtech got the designs from, with further investigation also possibly uncovering the links with China. Given that Xedra already has the predisposition to be suspicious of any seemingly abnormally advanced technology and is permanently paranoid that somebody might have discovered portal technology of their own, they dedicate a small portion of their resources to investigating the guns’ origins, a process that gets silenced by the cataclysm.
#### How’s All this Talk going to be implemented
The integration of the lore described is going to consist of two principle parts.
- [ ] Adding various snippets to transmit the lore to the player, should they find them. These snippets will range from electronic video logs to newspaper articles, advertisements, etc. This will additionally necessitate the reworking of any Rivtech-mentioning snippets already in the game to follow the new lore, should they conflict.
- [ ] Adding new locations, map extras, and location variants that tie into Rivtech. These will include, among others, locations where Rivtech was demonstrating their weapons before potential mass buyers, possible Rivtech company buildings, a Xedra testing area where they were investigating examples of Rivtech’s guns, and any other inclusions to establish Rivtech’s presence in pre-Cataclysm America. This will also require alterations to various item groups, including the stripping of all Rivtech guns from military sights and their introduction into commercial settings.
#### Magazines
- [ ] Rework all the Rivtech magazines on both a descriptive and implementive level. The current selection of Rivtech magazines possesses some fundamental mechanical issues, namely when it comes to their size and capacities. The in-game magazines, except for the drums, give no indication what style of magazine they are. If one is to assume that they’re G11-style top-fed single-stack box magazines, and keeping in mind that 8x40mm has been defined as being a 40x11x11mm box, then it means that a 50-round magazine should, at minimum, be 55 cm long, with a 100-round magazine coming in at 110 cm. This is clearly impractical for field use. The concept of being able to cycle box-shaped bullets effectively through a drum magazine is also impractical from a development standpoint. As such, I propose to remuddle the magazines to be clones of those used by the Russian VAG-73: a double-compartment magazine system that fundamentally consists of two double-stack magazines attached together. Magazines for standard use with caseless machineguns will either utilize an extended form of this magazine design or alternatively use a similar packet clip reloading system as was proposed for the H&K LMG G11.
#### fully-automatic firearms
- [ ] Rework the Rivtech RM-88 battle rifle. From all the current Rivtech guns, the RM88 bears the best representation of what a potential caseless rifle platform could look like, and it will be retooled into the base point of the Rivtech catalog, the rifle submitted for NGSW trials and later shipped around for use by 3rd party contractors. Heavier and slightly larger than most assault rifles, the weapon will stand on even footing with 7.62x51mm firearms like the FN SCAR-H and will be aided by access to far larger default magazine sizes, an integral gun scope, a suppressor, and access to a hyper-burst firing mode like the G11. This mode will effectively fire a number of rounds with the effective recoil of a single shot, leading to great accuracy when firing bursts.
- [ ] Rework the Rivtech RM 614 light machinegun. The RM614 is functionally an in-game representation of the preposed H&K LMG G11 concept: a caseless squad automatic weapon. As such, it will be reworked into Rivtech’s light machinegun submission for the NGSW trials, finding itself in a similar position to the battle rifle as it is sold to 3rd party contracters. The weapon will possess many of the same features as the battle rifle, including hyper-burst functionality, be fitted with a deployable bipod, and have its size upscaled to bring it in line with comparable SAWs.

#### semi-automatic and civilian firearms
- [ ] Rework the Rivtech RM103A magnum pistol. A self-loading handgun doesn’t make sense to contain a rifle cartridge like 8x40mm, and it is unreasonable to believe that Rivtech would invest resources into developing a hardened pistol action to contain such an energetic cartridge when they already have their hands full with bigger rifle projects. Bearing this in mind, a simpler alternative for Rivtech would see the weapon retooled to use a self-contained revolver-style action, inspired by the triple-chamber rotating cylinder of the LMG G11. Practically, the weapon would have been offered as an entry-level gun to civilians wishing to dip into the novelty of caseless weapon shooting, with the firearm filling the same niche as a big-bore revolver: a gun that people can buy to blast targets with, have fun with, and feel cool for owning.
- [ ] Rework the Rivtech RM11B scout rifle. The RM11B will be a post-NGSW development of the base battle rifle action, modified for precision service. In my mind, the system would be best served by a rework toward an expensive civilian hunting rifle. A separate and dedicated DMR/sniper variant could also be created to support Rivtech’s third-party militant offerings.
Rework the RM51 assault rifle. Given that the battle rifle will fill the role of Rivtech’s primary combat offering, the RM51 will be reworked into a civilian semi-automatic version of the battle rifle, retaining all the latter’s features save for its alternate firing modes.
- [ ] Rework the Rivtech RM2000 submachine gun. In offering a full-bore combat rifle as their primary militant arm, Rivtech would likely not see a reason to offer a submachine gun, which, regardless, doesn’t make sense on a pure taxonomic level as submachine guns fire pistol cartridges, something that 8x40mm plainly isn’t. What would make sense, however, is for Rivtech to produce a shortened version of their battle rifle and related civilian clone, creating a CQB carbine in the case of the former and an SBR in the case of the latter.
- [ ] Remove the Rivtech RM298 heavy machinegun. The concept of a heavy MG makes little functional sense given the flow of modern warfare, as pointed out in issue #33849, and it is unlikely that Rivtech would have ever made such a functionally useless design. As such, all references to the RM298 should be switched to the more "practical" RM614.
- [ ] Sort out ammo inconsistencies, including ironing out the armor-penetration numbers (probably making them parallel with.308 Winchester), updating damage values, removing 8x40mm HVP ammo (because nobody’s going to be putting depleted uranium in individual munitions), and conducting a general clean up.
#### Describe alternatives you've considered
Scrapping everything. All of this is, while my currently preferred vision after consultation and discussion with various people, ultimately still liable to possible changes should better thoughts come up or be presented.
#### Testing
Heavily WIP
#### Additional context
The link to the discord chat were all these ideas were originally born and workshopped

https://discord.com/channels/598523535169945603/610001025755119630/1097118030431006770

### CODE REVIEW

**Comment by John-Candlebury** [comment]
Damn this makes me feel terrible.

I see no point in changing the lore for our fictional gun brand from being black site projects to gimmick trash. I understand the reality check, but that part just seems insulting.

This doesnt seem inline with the consensus that was reached on the discord discussion, that one seems to be "Make them fully functional guns recovered through portal exploration".

**Comment by nexusmrsep** [comment]
I agree with John here. There shouldn't be reality checks on fictional stuff, if it's agreed that something is on the fictional side of the game. Trend toward realism should affect only things that have a real world reference, excluding things that cross that red line by design, which belong to the fiction or sci-fi and operate under fictional rules of whatever lore and balance applies.

**Comment by DoctorBoomstick** [comment]
To be fair, caseless guns do have a real-world parallel, said parallel being the H&K G11, which had all the issues outlined above. Caseless ammunition and related firearms simply have a number of inherent issues that are, frankly, still not going to be addressed, namely overheating and disastrous cookoff. Unfortunately, given mechanics and physics, these guns and 8x40mm are absurdly good for any technology one could ever realistically form, to the point where their only downside is rare ammo.

Also, nobody said that these guns were going to be singularly useless trash; far from it. They have, like all other weapons, advantages that are checked by practical downsides, such as high magazine sizes and improved ballistics over being bulkier and rarer. The military weapons are still going to retain utility; they’re just not going to be an all-in-one wonder weapon. As for the civilian arms, unfortunately, they’re representative of a company trying to make money and find ways of commercializing following a failed trial adoption, and such weapons would, by virtue of their issues and immature technological development, be range toys. At the end of the day, modern civilian electromagnetic rail guns, for example, while cool, are only ever practical as fun range toys.

Given some more development and R&D, they could have become viable weapons, but they are not because such a technology needs time to evolve, which these Rivtech guns don’t have. On the subject of their origins being alien portal weapons, my views on the matter are that such a bit of lore should be relegated to unreliable narration, serving to highlight the tension between Chinese and American relations and underlining Xedra’s paranoia that anyone could have possibly gotten their hands on portal technology. Having items derived from portal exploration and retrieval is a slippery slope to go down, and in established lore, which dictates the exorbitant costs associated with even visiting near-Earth dimensions and retrieving small samples, the concept that somebody could have popped into a sufficiently Earth-like dimension and made off with a bunch of caseless guns and ammo is rather questionable.

As I said, should other, better, and well-thought-out ideas be suggested that take into account the points above, then I have no issues with revising elements of the rework. My single standpoint on the matter at present is that, in a game that strives to realistically simulate everything from food cooking costs to what a nuclear-powered coffee machine could look like, these weapons, in their current forms, make little viable sense.

**Comment by Fris0uman** [comment]
No idea about the balance and physics of the thing but I think there's room in lore for rivtech guns being scifi super tech pulled from other world. Tweak the stats to make them nice super tech loot, put it in apropriate secret science/military sites, add some lore snippets and we're good to go

**Comment by nexusmrsep** [comment]
> To be fair, caseless guns do have a real-world parallel, said parallel being the H&K G11, which had all the issues outlined above. 

Of course they do, but that's not a reason to saying Rivtech guns need to be pulled down to the Earth realm from the sci-fi zone. One we call it sci-fi, the rest is "magic", and that's why we call it fiction anyway. Otherwise this would be equal to telling: hey we can't have laser guns in a sci-fi game, because we have have a real-world parallel in optical drives (DVDs), laser pointers, etc. and their power is very low, while military grade lasers are huge and use exorbitant amount of power.

> (...) such a bit of lore should be relegated to unreliable narration, (...). Having items derived from portal exploration and retrieval is a slippery slope to go down, and in established lore, (...) concept that somebody could have popped into a sufficiently Earth-like dimension and made off with a bunch of caseless guns and ammo is rather questionable.

> As I said, should other, better, and well-thought-out ideas be suggested that take into account the points above, then I have no issues with revising elements of the rework. 

If in your opinion it's the flaw of the sci-fi part of the lore (where there always is plenty of leeway anyway), then let me ask a question to approach the topic from the different side:

There is a need for rare powerful guns, that fit the role of hi-end scifi super tech weapon loot. Need to be top tier, player usable, sparkly, fun and games. Under the  environment and context of existing lore, and in the spirit and of current game state, what other lore would you propose for such brand of guns, what would be their sci-fi rationale, construction, mechanics etc. In other words: to fit the intended role - what would those guns be?

**Comment by DoctorBoomstick** [comment]
First and foremost, I wish to acknowledge that some of the points raised here are valid, and I recognize the concerns, as well as the fact that everybody with an issue isn’t dog piling with angry criticism and that many of the comments are quite reasonable. That being said, I would additionally like to say that the decisions made here weren’t, in any way, produced in a vacuum. Nor, for that matter, were they made without engaging in and listening to discussion and consultation with folks on the lore and functionality sides of things.
To everyone who's been commenting on discord and, to a lesser extent, here, stating that these changes nerf Rivtech guns into the ground, I would like to direct attention to the json itself and ask in what ways these guns are being degraded into trash. With respect to the battle rifle and light machinegun, their only realistic weaknesses stemming from the action taken in this PR have been:
- Increasing their, frankly, absurdly low dispersion numbers, with the XM88 retaining 10 less dispersion than the M2010 sniper rifle and the XM614 possessing somewhat less than the M240. Realistically, these changes are even offset by the fact that both of these guns now come kitted with an integral scope, so the downgrading is relatively low-scale in that respect.
- Rebalancing their weights and volumes, as they were quite wacky in some cases. While I don’t have the numbers at hand, one could carry something like 3–4 Rivtech light machineguns for every M249 SAW, with the battle rifle also having remarkably low weights and volumes attributed to it.
- removing the gigantically sized 100-round box and 500-round drum magazines. Following some rough calculations that I conducted to determine the lengths of the magazines, all of which were lacking defined lengths anyhow, the former would, assuming a single-stack box magazine configuration (the only realistically practical medium for housing rectangular caseless rounds), stand at 53.5 cm long, which, I think we can all agree, is impractically huge by any metric. The latter simply doesn’t make sense for a top-fed weapon.
- lowering their post-apocalyptic barter values, for reasons outlined in a dedicated JSON comment.
Everything else has remained the same or has indeed been improved. The weapons come kitted with a number of integral gunmods; the cartridge itself is slated to get a miner damage bonus as outlined in the statistics in the game documentation; and the weapons are liable to become more common following location spawn tweaks.
> There is a need for rare powerful guns, that fit the role of hi-end scifi super tech weapon loot. Need to be top tier, player usable, sparkly, fun and games. Under the environment and context of existing lore, and in the spirit and of current game state, what other lore would you propose for such brand of guns, what would be their sci-fi rationale, construction, mechanics etc. In other words: to fit the intended role - what would those guns be?
My standpoint on the matter is that we should gate such top-tier weapons behind forms of trans-dimensional meddling that have already been established to bring in a form of science fiction hardware, namely, the exodii. I see no reason to bring in another case of extreme portal jumping to justify in-game guns when the exodii are a faction with an established history of portal jumping that could have accessed all manner of other realities with all manner of other big fun weapons. It’s ok if all the end-game loot doesn’t come from labs and related black sites, and if every avenue of possible human endeavor didn’t get explored to the fullest and yield super-advanced technology. I believe that the concept of obtaining resources through portal research should be used sparingly and shrewdly for appropriate worldbuilding purposes. Otherwise, what’s to keep me from opening up a new PR and adding player-obtainable nuclear laptops, Amelia Airheart, and combat cyber hamsters, with the justification that they can be retrieved through portal shenanigans?
Rivtech guns don’t need to be the be all and end all of advanced weaponry, just because caseless guns = science fiction, and, while science fiction is very cool, there are some rules of basic physics that it must follow to be plausible. To this end, I suppose a happy medium could be a mixture of the proposed Rivtech rework with tastes of advanced portal origins mixed in.
I am not a particularly knowledgeable person on the matter of CDDA law, so my current thoughts are the result of ruminating and idea-forging, but perhaps the caseless guns could have their origins in alien dimensions, to a certain extent. The proposed lore building, which maintains that Rivtech pilfered caseless rifle designs from the Chinese, who themselves obtained them from portal research, could hold completely true rather than being an element of unreliable narration. Whether through dedicated large-scale portal exploration of their own or the intrusion of a small extradimensional force fleeing the blob, much like the Exodii, the Chinese could have gotten their hands on a number of alien, caseless weapons. These weapons, possibly obtained from an alternate reality Earth where the G11 program progressed and was allowed to mature into a practical weapon platform, could provide the initial stock of high-quality rifles and 8x40mm caseless munitions, which the Chinese could have been analyzing. These samples could, at some point, have been stolen or otherwise acquired by Rivtech, which sought to reverse-engineer the arms, a process that ultimately failed to figure out the mechanical functions of the advanced extradimensional rifles but prompted them to try experimental designs using the acquired 8x40mm munitions.

These prototype Rivtech designs, constructed following identifiable features of the weapons they were trying to reverse engineer, could form the military and civilian caseless weapons described within this PR’s description. The original extradimensional firearms, on the other hand, could be discoverable within labs, with Rivtech perhaps having tried working with Xedra to analyze the guns.

**Comment by tenmillimaster** [comment]
> Also, nobody said that these guns were going to be singularly useless trash; far from it.

Properly implemented, these would allow someone to pull off a 3(maybe 2 if boomstick wants to futz with the lengths and burst sizes) shot burst with no recoil applied til after the shots are fired.

They would also virtually eliminate fouling. The HITP propellant leaves virtually no residues.

There are reasons why a military would not want these guns, yet they might still be attractive to the player. 

As this is a draft PR, more work can be done to convey this to the player in the descriptions, as well as give it some 'sex appeal' to get that neuron activation for finding a rare gun. Please don't be harsh on the draft descriptions.

**Comment by halosammy** [comment]
Whatever my feelings may be on which way the pendulum should swing, this pull request neatly embodies the school of thought that the best compromises are the things no one likes. 

The realism camp gets its nerfs and the sci-fi camp gets Rivtech integrated into more recent lore. By all accounts, the company fits into the design document’s reference to shell companies that spun off from DARPA before XEDRA was organized. The Chinese angle is honestly unnecessary

With that in mind, something like this may be the only thing that stops people from returning to this issue.

**Comment by DoctorBoomstick** [comment]
> Properly implemented, these would allow someone to pull off a 3(maybe 2 if boomstick wants to futz with the lengths and burst sizes) shot burst with no recoil applied til after the shots are fired.
> 
> They would also virtually eliminate fouling. The HITP propellant leaves virtually no residues.

These are my sentiments, precisely. A recoilless hyper-burst system and mainly non-fouling action are both on the cards, with their only reason for not being in the PR at present being that I have yet to find out how to model switching to a less-recoiling fire mode. The alterations to the weapons description to point towards their utility while still indicating their impracticability for military trials are a small matter to change, and the references to the notion that they would have faded into obscurity could be trimmed down as a compromise. So, ladies and gentlemen, I suggest holding your horses a bit when stating that Rivtech is going to be drastically downgraded. Many changes have yet to come in, and the final product is likely to retain much player utility.
> This is the part where I include my feelings and say that don’t see the point in fixing what isn’t broken. It’s not like farming 8x40mm ammo is a viable strategy when it comes to actually surviving the game without using the debug console. Although I acknowledge my bias in frankly enjoying the feeling of going nuts after I save up enough HVP rounds over the long travels.

My philosophy is that justifying the existence of science fiction elements based on the notion that they existed and were classified as science fiction before the shift towards CDDA’s more realistic leanings, and that such elements should have game lore adjusted to fit them rather than changing in their implementation, is rather silly. If we follow this school of thought, we could have handwaved away bionics as simply coming from portal exploration rather than shifting them to technology brought to Earth by the Exodii. When it comes to Rivtech guns, should they be implemented today in their present form, they would likely never make it past a proposed PR, for they are, bluntly, overpowered for what they purport to be, standing in as the best of all worlds, a no-issue platform.
With respect to the Chinese leanings in particular, I personally liked the suggestion as it helped to expand pre-Cataclysm lore further in ways that made sense and were interesting. It expands the notion that portal-related investigations and happenings weren’t only relegated to the US and places the world of CDDA against a larger international backdrop. I like it better than the idea of the guns being extracted from portals through US research, as, firstly, such a happening has already taken place, and it personally feels a tad lazy to hash out the same snippet of lore to justify this particular case.
Secondly, given Xedra’s philosophy of frequent but small-scale portal ventures, the idea that they could have crossed to a completely different reality on Earth to extract highly-advanced guns is questionable, as the established resource requirement would be nothing short of titanically massive. Having these weapons come into our reality through the Chinese, whether through much larger and much less common portal explorations of their own or, my personal favorite, ending up with a small party of extradimensional beings carrying the guns, either as these individuals are escaping from the blob themselves or on a portal exploration from an alternate reality Earth, just sounds more interesting.
As for Rivtech being a shell corporation spun off from DARPA, there is no reason why this couldn’t be implemented within the landscape of this lore. Perhaps, after getting wind of the Chinese coming into possession of seemingly unusually advanced firearms and correctly suspecting their extradimensional origins, Xedra could have arranged for the guns to be clandestinely smuggled out of Chinese hands and brought to the US, where they could be turned over to Rivtech for reverse engineering.

**Comment by tenmillimaster** [comment]
> Secondly, given Xedra’s philosophy of frequent but small-scale portal ventures, the idea that they could have crossed to a completely different reality on Earth to extract highly-advanced guns is questionable, as the established resource requirement would be nothing short of titanically massive. Having these weapons come into our reality through the Chinese, whether through much larger and much less common portal explorations of their own or, my personal favorite, ending up with a small party of extradimensional beings carrying the guns, 

Though this does not preclude your inclusions of the guns as having a Chinese extra dimensional origin, I should note that the basis you're presenting here is not formally considered canon, but is an idea I presented that I felt was a logical extension of our existing lore/portal physics. I presented it because I felt it necessary to preserve the status quo and help explain some of the principles laid out in the design document, such as, "question why the thing you are making has to be sci-fi". Unless Kevin or any other senior dev managing the design document signs off on that, my understanding is that the current lore is that any tech can be extracted from portals, pre-cataclysm, as long as it still complies with the design document.

**Comment by halosammy** [comment]
`As for Rivtech being a shell corporation spun off from DARPA, there is no reason why this couldn’t be implemented within the landscape of this lore. Perhaps, after getting wind of the Chinese coming into possession of seemingly unusually advanced firearms and correctly suspecting their extradimensional origins, Xedra could have arranged for the guns to be clandestinely smuggled out of Chinese hands and brought to the US, where they could be turned over to Rivtech for reverse engineering. `

I’m actually quite down for this angle.

File: data/json/items/magazine/8x40mm.json [hunk 1]
```diff
@@ -1,21 +1,75 @@
...
-    "id": "8x40_100_mag",
+    "id": "8x40_44_mag",
     "looks_like": "stanag30",
     "type": "MAGAZINE",
-    "name": { "str": "8x40mm 100-round magazine" },
-    "description": "A 100-round box magazine for use with Rivtech 8x40mm caseless firearms.  Designated as RMGB100.",
-    "weight": "150 g",
-    "volume": "750 ml",
+    "name": { "str": "8x40mm 44-round magazine" },
+    "description": "a shortened 44-round box magazine for Rivtech 8x40mm caseless firearms, built from translucent polymer and designed to lock into place above a weapon’s barrel and cycle cartridges perpendicularly downwards into a firearm’s chamber.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/8x40mm.json:7]:
```suggestion
    "description": "A shortened 44-round box magazine for Rivtech 8x40mm caseless firearms, built from translucent polymer and designed to lock into place above a weapon’s barrel and cycle cartridges perpendicularly downwards into a firearm’s chamber.",
```

File: data/json/items/magazine/8x40mm.json [hunk 2]
```diff
@@ -1,21 +1,75 @@
...
+    "ammo_type": [ "8x40mm" ],
+    "flags": [ "MAG_COMPACT" ],
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "8x40mm": 44 } } ]
+  },
+  {
+    "id": "8x40_56_mag",
+    "looks_like": "stanag30",
+    "type": "MAGAZINE",
+    "name": { "str": "8x40mm 56-round magazine" },
+    "description": "a 56-round box magazine for Rivtech 8x40mm caseless firearms, built from translucent polymer and designed to lock into place above a weapon’s barrel and cycle cartridges perpendicularly downwards into a firearm’s chamber.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/8x40mm.json:25]:
```suggestion
    "description": "A 56-round box magazine for Rivtech 8x40mm caseless firearms, built from translucent polymer and designed to lock into place above a weapon’s barrel and cycle cartridges perpendicularly downwards into a firearm’s chamber.",
```

**Comment by halosammy** [comment]
Jeez, this is still going on? If you’re still settling on a lore rework to go with the new items, you can just make it one of the DARPA shell companies that XEDRA eventually reabsorbed. No need to overthink it.

**Comment by tenmillimaster** [comment]
Heya Boomstick, are ya still working on this?

**Comment by tenmillimaster** [comment]
Ah Shit, I'm sorry Boomstick, I didn't realize checking and rebasing your branch would start merging stuff to it!

**Pull request closed without merging**